### ISSUE TITLE
t0 definition in models

### ISSUE THREAD

**Issue by JannisNe** [sncosmo/sncosmo#204]
I am simulating light curves using different models and realized that t0 is in some cases defined as the peak time (e.g. SALT2) and others it's the explosion time (e.g. nugget-sn1a).
This is complicated and makes the coding harder and I wanted to know if it would be possible to have a more consistent definition of t0.
Alternatively could it be possible to have an expected rising time?

**Comment by kbarbary** [comment]
I agree. I've experienced this annoyance when using sncosmo for simulations.

To explain the rationale for how it currently works: making `t0` correspond to "maximum" for all models would be problematic, because the time of maximum is different in different bands (or different wavelengths). Furthermore, adding dust to a model changes the time of maximum (slightly). Making `t0` correspond to explosion probably would have worked fine: explosion time is the same in all bands and is not affected by things like dust. However, I think this would have caused some confusion when using a popular model like SALT2: `t0` in sncosmo's SALT2 would have a different definition than that used in the "standard" SALT2 fitter `snfit`.

Also, at this point it will be difficult to change the phase definition of built-in `Source`s without breaking existing code. However, a compromise might be to add a method to the `Model` class that allows you to set the time-of-explosion, regardless of how phase / `t0` is defined for that particular model. Something like `model.set_mintime(55000.)`. It should be pretty easy to implement this and should make simulation tasks simpler.

**Comment by rbiswas4** [comment]
> Also, at this point it will be difficult to change the phase definition of built-in Sources without breaking existing code. However, a compromise might be to add a method to the Model class that allows you to set the time-of-explosion, regardless of how phase / t0 is defined for that particular model. Something like model.set_mintime(55000.)

Are you proposing to define `Model.mintime = t0 - 30.0 * (1+z) ` for SALT2  and time of explosion for Nugent templates? Will this be done  in SNCosmo itself or will these be user defined?

I think it would be nice to have SNCosmo define (at least some default value) for `mintime` .

**Comment by kbarbary** [comment]
@rbiswas4 Not sure if this is what you're asking, but `Model.mintime()` already exists:

```python
In [1]: import sncosmo

In [2]: model = sncosmo.Model('salt2')

In [3]: model.mintime()
Out[3]: -20.0

In [4]: model.set(z=0.5, t0=55000.)

In [5]: model.mintime()
Out[5]: 54970.0
```

I'm simply proposing adding a convenience function to set `t0` such that `Model.mintime()` achieves some desired value.

**Comment by rbiswas4** [comment]
@kbarbary Ah yes, I did not realize this is the same variable. But that `mintime` is a statement of what time range the model is defined to and not the `explosion time` and it might be good to separate these.
